pkg/rc/replication_controller.go

```diff
@@ -614,3 +615,82 @@ func (rc *replicationController) unschedule(txn *auditingTransaction, node types
 	txn.RemoveNode(node)
 	return nil
 }
+
+func (rc *replicationController) transferNodes(ineligibleCurrent []types.NodeName) error {
+	var err error
+	if rc.AllocationStrategy == fields.CattleStrategy {
+		err = rc.transferCattleNodes(ineligibleCurrent)
+	} else {
+		err = rc.transferPetNodes(ineligibleCurrent)
+	}
+	if err != nil {
+		return util.Errorf("Unable to transfer nodes: %s", err)
+	}
+
+	return nil
+}
+
+func (rc *replicationController) transferCattleNodes(ineligibleCurrent []types.NodeName) error {
+	if len(ineligibleCurrent) < 1 {
```

Comment:
Note: we may change this if we decide to bring this node allocation functionality to `addPods()` since there would be no current ineligible nodes in that case